Bottom: af8225c629e4b76b0fac62ec18d33e36d90f3130
Top:    fd540344fb8f779006ef57aa881b40bf5824785b
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-02-17 15:13:47 +0800

Refresh of fixup

---

diff --git a/drivers/net/virtio_net.c b/drivers/net/virtio_net.c
index 0bd1101..05474b0 100644
--- a/drivers/net/virtio_net.c
+++ b/drivers/net/virtio_net.c
@@ -813,7 +813,8 @@ static int add_recvbuf_small(struct virtnet_info *vi, struct receive_queue *rq,
 
 	buf = (char *)page_address(alloc_frag->page) + alloc_frag->offset;
 	get_page(alloc_frag->page);
-	sg_init_one(rq->sg, buf + NET_IP_ALIGN + xdp_headroom, GOOD_PACKET_LEN);
+	sg_init_one(rq->sg, buf + NET_IP_ALIGN + xdp_headroom,
+		    hdr_len + GOOD_PACKET_LEN);
 	err = virtqueue_add_inbuf(rq->vq, rq->sg, 1, buf, gfp);
 	if (err < 0)
 		put_page(virt_to_head_page(buf));
